src/LegacyRepContract.sol

```diff
@@ -0,0 +1,40 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/libraries/token/VariableSupplyToken.sol';
+import 'ROOT/libraries/token/ERC20.sol';
+
+
+contract LegacyRepContract is VariableSupplyToken {
+    event FundedAccount(address indexed _branch, address indexed _sender, uint256 _repBalance, uint256 _timestamp);
+
+    uint256 private constant FAUCET_AMOUNT = 47 ether;
+
+    string public constant name = "Reputation";
+    string public constant symbol = "REP";
+    uint256 public constant decimals = 18;
+
+    address private creator;
+
+    function LegacyRepContract() {
+        // This is to confirm we are on test net
+        address _foundationRepAddress = address(0xe94327d07fc17907b4db788e5adf2ed424addff6);
+        uint256 size;
+        assembly { size := extcodesize(_foundationRepAddress) }
+        require(size == 0);
+        // Store the creator for use in automated testing funding
+        creator = msg.sender;
+    }
+
+    function faucet() public returns (bool) {
+        require(balanceOf(msg.sender) == 0);
+        mint(msg.sender, FAUCET_AMOUNT);
+        FundedAccount(this, msg.sender, FAUCET_AMOUNT, block.timestamp);
+        return true;
+    }
+
+    // Function for automated testing
+    function testFund(uint256 _amount) public returns (bool) {
+        require(msg.sender == creator);
+        mint(msg.sender, _amount);
+    }
```

Comment:
Can you help me understand this?  Why not just call `faucet`?  If its because you want the ability to ask for a certain amount, we can just have an overload of `faucet(uint256)` that takes in an amount.  There isn't any real need to secure it against someone asking for a lot.  Maybe have a cap at something like 2^128 just so someone can't ask for so much REP that the totalSupply caps out (would require them calling that method 2^128 times to cap out totalSupply).

Reply:
Yea given we don't care about restricting it I'll just make this take an argument as you suggested